### PR TITLE
[IMP] account: Remove regex matching on aml to retrieve statement par…

### PR DESCRIPTION
**Pull request by smetl**
…tner

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_bank_statement_line.py [hunk 1]
```diff
@@ -633,46 +633,6 @@ def _retrieve_partner(self):
             if partner and rec_model._is_applicable_for(self, partner):
                 return partner
 
-        # Retrieve the partner from statement line text values.
-        st_line_text_values = self._get_st_line_strings_for_matching()
-        unaccent = get_unaccent_wrapper(self._cr)
```
Comment (william-andre) [inline addons/account/models/account_bank_statement_line.py:638]: Seems like an unused import now?

**Review by william-andre — COMMENTED** [review]
```
[IMP] account: do not grep partners for statement matching

Doing a regex search forces a sequential scan on too many rows.
Because this is a function called often during the bank reconciliation,
it is better to prioritize a faster flow than an automatic detection that is 
likely to not return anything anyways.
```

**Comment by smetl** [comment]
@william-andre done

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**